### PR TITLE
vmi controller: Fix UID comparison

### PR DESCRIPTION
**Pull request by oshoval**
**What this PR does / why we need it**:
In case claim already exists,
need to compare to the expected UID
which might belong either to the VM or VMI,
depending if the VMI has owner VM or standalone.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by oshoval** [comment]
/hold

depends on https://github.com/kubevirt/ipam-extensions/pull/37

File: pkg/vminetworkscontroller/vmi_controller.go [hunk 1]
```diff
@@ -117,8 +117,8 @@ func (r *VirtualMachineInstanceReconciler) Reconcile(
 						return controllerruntime.Result{}, fmt.Errorf("let us be on the safe side and retry later")
 					}
 				}
-				if len(existingIPAMClaim.OwnerReferences) == 1 && existingIPAMClaim.OwnerReferences[0].UID == vm.UID {
-					r.Log.Info("found existing IPAMClaim belonging to this VM, nothing to do", "VM UID", vm.UID)
+				if len(existingIPAMClaim.OwnerReferences) == 1 && existingIPAMClaim.OwnerReferences[0].UID == ownerInfo.UID {
```
Comment (qinqon) [inline pkg/vminetworkscontroller/vmi_controller.go:120]: We should add some e2e coverage here, this should have fail at CI.

**Review by qinqon — CHANGES_REQUESTED** [review]

File: pkg/vminetworkscontroller/vmi_controller.go (hunk 1 above)
Comment (maiqueb) [reply pkg/vminetworkscontroller/vmi_controller.go:120]: I agree. Please add a test for this. 

Still, good catch. This was a leftover of when I introduced VMI support.

**Comment by oshoval** [comment]
/hold cancel

rebased

**Comment by qinqon** [comment]
/hold cancel 
all good

**Comment by oshoval** [comment]
Will create a tag bit later so it will reach D/S

**Pull request merged**